### PR TITLE
Add api auth

### PR DESCRIPTION
**Pull request by nelsonkopliku**
This PR
- Loads react app only if logged in, avoiding unnecessary requests on the login page itself
- Adds the ability to generate a deterministic API key for a Trento installation
- The installation usecase exposes the possibility to retrieve that API key
- Added an api controller to expose the api key to the frontend
- added authentication middlewares (plugs) supporting api keys
- secured APIs to make them accessible only when logged in

the api key will be copied and used during trento agents installation.

After logging in and having copied the cookie

```
curl --request GET 'http://localhost:4000/api/installation/api-key' \ 
--header 'Cookie: <TRENTO_COOKIES> | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   172  100   172    0     0   2606      0 --:--:-- --:--:-- --:--:--  2646
{
  "apiKey": "The API key"
}
```

The you can execute 
```
curl --location --request GET 'http://localhost:4000/api/test' \
--header 'X-Trento-apiKey: <the generated api key>' \
--header 'Content-Type: application/json'
```
(try it also without auth)

Next: 
- Provide a UI to grab the api key
- extend the agent to accept the api key during startup and send it along with the discoveries
- put the routes in the proper pipeline
- make internal routes not accessible at ingress level
- Trento web https

### CODE REVIEW

**Review by fabriziosestito — APPROVED** [review]
LGTM! Just one comment about naming consistency

**Comment by rtorrero** [comment]
> This PR
> 
>     * Loads react app only if logged in, avoiding unnecessary requests on the login page itself
Maybe this means we don't need to do the:
```
var trento = document.getElementById('trento');
if (trento) {
  render(<App />, trento);
}
```
we do in [assets/js/trento.js:71](https://github.com/trento-project/web/blob/a087e6e854758d5be04d7a037f3fdd9b6e4948e6/assets/js/trento.jsx#L71)

**Comment by rtorrero** [comment]
The e2e tests are apparently no longer showing `hana_cluster_22` and `hana_cluster_3` from the fixtures as healthy; not sure why it started failing here

File: lib/trento_web/controllers/installation_controller.ex [hunk 1]
```diff
@@ -0,0 +1,11 @@
+defmodule TrentoWeb.InstallationController do
+  use TrentoWeb, :controller
+
+  def get_api_key(conn, _) do
+    key = Trento.Installation.get_api_key()
+
+    conn
+    |> put_status(:ok)
+    |> json(%{apiKey: key})
```
[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply lib/trento_web/controllers/installation_controller.ex]: Yup changed.

**Pull request merged**